.github/workflows/ci.yml

```diff
@@ -97,30 +118,21 @@ jobs:
           if [ "${{ matrix.backend }}" = "vulkan" ]; then
             brew install molten-vk vulkan-headers
           fi
-      # Regular tests
-      - run: just ci-test ${{ matrix.backend }}
-      - run: just test-doc ${{ matrix.backend }}
-      # Test MSRV
+      # Setup rust version
+      - run: rustup update stable && rustup default stable
+        if: matrix.rust-version == 'stable'
       - name: Read MSRV
         id: msrv
         run: echo "value=$(sed -nE 's/^rust-version[[:space:]]*=[[:space:]]*"([^"]+)".*/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
+        if: matrix.rust-version == 'msrv'
```

Comment:
not a biggie, but usually i put `if` condition as the very first line of the block:

```yaml
- if: ...
  name: ...
  ...
```

This way it reads more naturally - i.e. conditionality of a block is apparent from the start